### PR TITLE
Effectively remove `RNGestureHandlerEnabledRootView`

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

`tryInitializeHandlerForReactRootView` was causing crashes, especially with combination with CodePush. `RNGestureHandlerEnabledRootView` was deprecated in version 2.0.0 in favor of `GestureHandlerRootView` component and this PR removes implementation of `RNGestureHandlerEnabledRootView` and code responsible for interactions with it.
Adds a new page in docs explaining how to migrate to `GestureHandlerRootView`.

Potentially fixes https://github.com/software-mansion/react-native-gesture-handler/issues/1675.

## Test plan

Checked on the Example app

### CODE REVIEW

**Comment by LFMAKER** [comment]
@j-piasecki Hi, very happy to see that you found a way to solve this problem. When can this be merged?

**Comment by j-piasecki** [comment]
@LFMAKER It should be merged after @kmagiera gives it a green light, but hopefully soon.

File: docs/docs/guides/migrating-off-rnghenabledroot.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+-      @Override
+-      protected ReactRootView createRootView() {
+-       return new RNGestureHandlerEnabledRootView(MainActivity.this);
+-      }
+-    };
+-  }
+}
+```
+
+## Update your JS code
```
Comment (kmagiera) [inline docs/docs/guides/migrating-off-rnghenabledroot.md:29]: Add a note that react navigation already creates root container in which case adding rootview component won't be necessary

**Review by kmagiera — APPROVED** [review]

File: docs/docs/guides/migrating-off-rnghenabledroot.md (hunk 1 above)
Comment (j-piasecki) [reply docs/docs/guides/migrating-off-rnghenabledroot.md:33]: Added in https://github.com/software-mansion/react-native-gesture-handler/pull/1936/commits/5cb102f1b2c6a5a398ce5a79629e2831a0575a26.

**Pull request merged**